src/coinbase/webhook.ts

```diff
@@ -185,10 +185,13 @@ export class Webhook {
     notificationUri,
     eventTypeFilter,
   }: UpdateWebhookOptions): Promise<Webhook> {
+    const finalNotificationUri = notificationUri ?? this.getNotificationURI();
+    const finalEventTypeFilter = eventTypeFilter ?? this.getEventTypeFilter();
+
     const result = await Coinbase.apiClients.webhook!.updateWebhook(this.getId()!, {
-      notification_uri: notificationUri,
+      notification_uri: finalNotificationUri,
       event_filters: this.getEventFilters()!,
-      event_type_filter: eventTypeFilter,
+      event_type_filter: finalEventTypeFilter,
     });
```

Comment:
Are all of those props required on the request? What about not sending it, if user hasn't sent on the SDK function param? It would be ideal to the API request to mimic what user intended, this could also possibly prevent issues in the future.

Reply:
because on the CDP side this is a `PUT` endpoint, so we have to provide all these parameters when calling it. These parameters are as good as required, although not labeled as "required".